linux/keyman-config/keyman_config/install_kmp.py

```diff
@@ -250,7 +250,7 @@ def install_keyboards_to_ibus(keyboards, packageDir):
 			# install all kmx for first lang not just packageID
 			for kb in keyboards:
 				kmx_file = os.path.join(packageDir, kb['id'] + ".kmx")
-				if "languages" in kb:
+				if "languages" in kb and len(kb["languages"]) > 0:
 					logging.debug(kb["languages"][0])
 					keyboard_id = "%s:%s" % (kb["languages"][0]['id'], kmx_file)
 				else:
```

Comment:
@mcdurdin said if languages wasn't set, to have the following fallbacks:

- Fallback to .kmx language (`ethnologuelanguages`, `language` and `windowslanguages` stores, ideally), 
- Then fallback to user's default language
- Finally if that can't be found, fallback to `en`

fwiw, ibus seems ok not having a language. ge.kmp installed as "Other (General)"

Reply:
Given that ibus supports "Other" that makes sense. On Windows, you don't have that option so we fall back to user's default language.

Re kmx agree with @glasseyes. However, in the future, for many older keyboards, the kmx has good language data, so worth considering for an update.